docs/rfcs/004-multi-file-diff.md

```diff
@@ -0,0 +1,72 @@
+# Multi-file Diffs
+
+## Status
+
+Proposed
+
+## Summary
+
+Users can select multiple files in unstaged changes and staged changes pane and see the diffs of selected files in one view, akin to the [`Files changed` tab in pull requests on github.com](https://github.com/atom/github/pull/1753/files).
+
+## Motivation
+
+So that users can view a full set of changes with more context before staging or committing those changes.
+
+The ability to display multiple diffs in one view will also serve as a building block for the following planned features:
+- [commit pane item](#1655) where it shows all changes in a single commit
+- [new PR review flow](https://github.com/atom/github/blob/master/docs/rfcs/003-pull-request-review.md) that shows all changed files proposed in a PR
+
+
+## Explanation
+
+#### Mock-ups coming soon
+
+
+#### Unstaged Changes pane
+- User can `cmd+click` and select multiple files from the list of unstaged changes, and the pane on the left (see multi-file diff section below) will show diffs of the selected files. That pane will continue to reflect any further selecting/unselecting on the Unstaged Changes pane.
+- Once there is at least one file selected, `Stage All` button should be worded as `Stage Selected`.
+
+#### Staged Changes pane
+- Same behavior as Unstaged Changes pane.
+- Once there is at least one file selected, `Unstage All` button should be worded as `Unstage Selected`.
+
+![staged changes](https://user-images.githubusercontent.com/378023/47497740-0a0b3200-d896-11e8-85af-7c644af9ca37.png)
```

Comment:
Added some mockups, but I'm not so sure about this part:

![screen shot 2018-10-25 at 8 48 48 pm](https://user-images.githubusercontent.com/378023/47498275-bdc0f180-d897-11e8-9ac5-f1d177d4c165.png)

The buttons to stage file and hunks look kinda weird when they are close together. Maybe we could only show them on a file when the file is collapsed.

Reply:
How does it look if the buttons were the same width? Or is that just more confusing?